Upgrade to resteasy 3.6.2.Final broke tck

Upgrade to resteasy 3.6.2.Final https://github.com/smallrye/smallrye-opentracing/pull/34  broke tck with the following error. Strange is that CI is passing fine: https://github.com/smallrye/smallrye-opentracing/pull/45

```
16:20:54,612 ERROR [org.jboss.msc.service.fail] (ServerService Thread Pool -- 26) MSC000001: Failed to start service jboss.deployment.unit."opentracing.war".undertow-deployment: org.jboss.msc.service.StartException in service jboss.deployment.unit."opentracing.war".undertow-deployment: java.lang.NoSuchFieldError: SERVER_SENT_EVENTS_TYPE
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:81)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1985)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1487)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1378)
	at java.lang.Thread.run(Thread.java:748)
	at org.jboss.threads.JBossThread.run(JBossThread.java:485)
Caused by: java.lang.NoSuchFieldError: SERVER_SENT_EVENTS_TYPE
	at org.jboss.resteasy.core.ResourceMethodInvoker.isSseResourceMethod(ResourceMethodInvoker.java:161)
	at org.jboss.resteasy.core.ResourceMethodInvoker.<init>(ResourceMethodInvoker.java:146)
	at org.jboss.resteasy.core.ResourceMethodRegistry.processMethod(ResourceMethodRegistry.java:349)
	at org.jboss.resteasy.core.ResourceMethodRegistry.register(ResourceMethodRegistry.java:276)
	at org.jboss.resteasy.core.ResourceMethodRegistry.addResourceFactory(ResourceMethodRegistry.java:227)
	at org.jboss.resteasy.core.ResourceMethodRegistry.addResourceFactory(ResourceMethodRegistry.java:199)
	at org.jboss.resteasy.core.ResourceMethodRegistry.addResourceFactory(ResourceMethodRegistry.java:185)
	at org.jboss.resteasy.core.ResourceMethodRegistry.addResourceFactory(ResourceMethodRegistry.java:162)
	at org.jboss.resteasy.core.ResourceMethodRegistry.addPerRequestResource(ResourceMethodRegistry.java:79)
	at org.jboss.resteasy.spi.ResteasyDeployment.registration(ResteasyDeployment.java:498)
	at org.jboss.resteasy.spi.ResteasyDeployment.startInternal(ResteasyDeployment.java:281)
	at org.jboss.resteasy.spi.ResteasyDeployment.start(ResteasyDeployment.java:88)
	at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.init(ServletContainerDispatcher.java:119)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.init(HttpServletDispatcher.java:36)
	at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:117)
	at org.wildfly.extension.undertow.security.RunAsLifecycleInterceptor.init(RunAsLifecycleInterceptor.java:78)
	at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:103)
	at io.undertow.servlet.core.ManagedServlet$DefaultInstanceStrategy.start(ManagedServlet.java:303)
	at io.undertow.servlet.core.ManagedServlet.createServlet(ManagedServlet.java:143)
	at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:583)
	at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:554)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at org.wildfly.extension.undertow.security.SecurityContextThreadSetupAction.lambda$create$0(SecurityContextThreadSetupAction.java:105)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at io.undertow.servlet.core.DeploymentManagerImpl.start(DeploymentManagerImpl.java:596)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService.startContext(UndertowDeploymentService.java:97)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:78)
	... 8 more

16:20:54,614 ERROR [org.jboss.as.controller.management-operation] (management-handler-thread - 1) WFLYCTL0013: Operation ("add") failed - address: ([("deployment" => "opentracing.war")]) - failure description: {"WFLYCTL0080: Failed services" => {"jboss.deployment.unit.\"opentracing.war\".undertow-deployment" => "java.lang.NoSuchFieldError: SERVER_SENT_EVENTS_TYPE
    Caused by: java.lang.NoSuchFieldError: SERVER_SENT_EVENTS_TYPE"}}
16:20:54,616 ERROR [org.jboss.as.server] (management-handler-thread - 1) WFLYSRV0021: Deploy of deployment "opentracing.war" was rolled back with the following failure message: 
{"WFLYCTL0080: Failed services" => {"jboss.deployment.unit.\"opentracing.war\".undertow-deployment" => "java.lang.NoSuchFieldError: SERVER_SENT_EVENTS_TYPE
    Caused by: java.lang.NoSuchFieldError: SERVER_SENT_EVENTS_TYPE"}}
```

@michalszynkiewicz have you seen this when updating?

Not pinning jaxrs in parent to 2.1 didn't resolve this. Wondering whether the following code from TCK is causing the issue. The json-provider:2.9.0 depends on jaxrs 2.0.
```
public static WebArchive createDeployment() {

        File[] files = Maven.resolver()
            .resolve(
                "io.opentracing:opentracing-api:0.31.0",
                "com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.0"
            )
            .withTransitivity().asFile();

        WebArchive war = ShrinkWrap.create(WebArchive.class, "opentracing.war")
            .addPackages(true, OpenTracingClientBaseTests.class.getPackage())
            .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml")
            .addAsLibraries(files);

        return war;
    }
```

https://github.com/eclipse/microprofile-opentracing/blob/master/tck/src/main/java/org/eclipse/microprofile/opentracing/tck/OpenTracingBaseTests.java#L77

Here is content of the lib folder when running tck:
```
/WEB-INF/lib/opentracing-api-0.31.0.jar
/WEB-INF/lib/jackson-jaxrs-json-provider-2.9.0.jar
/WEB-INF/lib/jackson-jaxrs-base-2.9.0.jar
/WEB-INF/lib/jackson-core-2.9.0.jar
/WEB-INF/lib/jackson-databind-2.9.0.jar
/WEB-INF/lib/jackson-module-jaxb-annotations-2.9.0.jar
/WEB-INF/lib/jackson-annotations-2.9.0.jar
/WEB-INF/lib/extension.jar
/WEB-INF/lib/smallrye-opentracing-1.2.2-SNAPSHOT.jar
/WEB-INF/lib/microprofile-opentracing-api-1.2.1.jar
/WEB-INF/lib/org.osgi.annotation.versioning-1.0.0.jar
/WEB-INF/lib/microprofile-config-api-1.1.jar
/WEB-INF/lib/opentracing-jaxrs2-0.3.1.jar
/WEB-INF/lib/opentracing-noop-0.31.0.jar
/WEB-INF/lib/opentracing-web-servlet-filter-0.2.0.jar
/WEB-INF/lib/opentracing-util-0.31.0.jar
/WEB-INF/lib/opentracing-concurrent-0.2.0.jar
/WEB-INF/lib/opentracing-interceptors-0.0.4.jar
/WEB-INF/lib/javaee-api-7.0.jar
/WEB-INF/lib/javax.mail-1.5.0.jar
/WEB-INF/lib/opentracing-tracerresolver-0.1.5.jar
/WEB-INF/lib/resteasy-cdi-3.6.2.Final.jar
/WEB-INF/lib/jboss-jaxrs-api_2.1_spec-1.0.2.Final.jar
/WEB-INF/lib/validation-api-1.1.0.Final.jar
/WEB-INF/lib/resteasy-jaxrs-3.6.2.Final.jar
/WEB-INF/lib/jboss-jaxb-api_2.3_spec-1.0.0.Final.jar
/WEB-INF/lib/reactive-streams-1.0.2.jar
/WEB-INF/lib/jboss-annotations-api_1.2_spec-1.0.0.Final.jar
/WEB-INF/lib/activation-1.1.1.jar
/WEB-INF/lib/httpclient-4.5.4.jar
/WEB-INF/lib/httpcore-4.4.7.jar
/WEB-INF/lib/commons-logging-1.2.jar
/WEB-INF/lib/commons-codec-1.10.jar
/WEB-INF/lib/commons-io-2.5.jar
/WEB-INF/lib/jcip-annotations-1.0.jar
/WEB-INF/lib/jboss-logging-3.3.1.Final.jar
/WEB-INF/lib/microprofile-opentracing-tck-1.2.1.jar
/WEB-INF/lib/testng-6.9.9.jar
/WEB-INF/lib/junit-4.10.jar
/WEB-INF/lib/hamcrest-core-1.1.jar
/WEB-INF/lib/jcommander-1.48.jar
/WEB-INF/lib/ant-1.7.0.jar
/WEB-INF/lib/ant-launcher-1.7.0.jar
/WEB-INF/lib/snakeyaml-1.15.jar
/WEB-INF/lib/bsh-2.0b4.jar
/WEB-INF/lib/arquillian-testng-container-1.4.1.Final.jar
/WEB-INF/lib/arquillian-testng-core-1.4.1.Final.jar
/WEB-INF/lib/arquillian-test-api-1.4.1.Final.jar
/WEB-INF/lib/arquillian-core-api-1.4.1.Final.jar
/WEB-INF/lib/arquillian-test-spi-1.4.1.Final.jar
/WEB-INF/lib/arquillian-core-spi-1.4.1.Final.jar
/WEB-INF/lib/arquillian-container-test-api-1.4.1.Final.jar
/WEB-INF/lib/shrinkwrap-descriptors-api-base-2.0.0.jar
/WEB-INF/lib/arquillian-container-test-spi-1.4.1.Final.jar
/WEB-INF/lib/arquillian-container-spi-1.4.1.Final.jar
/WEB-INF/lib/arquillian-core-impl-base-1.4.1.Final.jar
/WEB-INF/lib/arquillian-test-impl-base-1.4.1.Final.jar
/WEB-INF/lib/arquillian-container-impl-base-1.4.1.Final.jar
/WEB-INF/lib/arquillian-config-api-1.4.1.Final.jar
/WEB-INF/lib/arquillian-config-impl-base-1.4.1.Final.jar
/WEB-INF/lib/arquillian-config-spi-1.4.1.Final.jar
/WEB-INF/lib/shrinkwrap-descriptors-spi-2.0.0.jar
/WEB-INF/lib/arquillian-container-test-impl-base-1.4.1.Final.jar
/WEB-INF/lib/shrinkwrap-impl-base-1.2.6.jar
/WEB-INF/lib/shrinkwrap-spi-1.2.6.jar
/WEB-INF/lib/shrinkwrap-api-1.2.6.jar
/WEB-INF/lib/shrinkwrap-resolver-impl-maven-2.2.6.jar
/WEB-INF/lib/shrinkwrap-resolver-spi-maven-2.2.6.jar
/WEB-INF/lib/shrinkwrap-resolver-spi-2.2.6.jar
/WEB-INF/lib/aether-api-1.0.0.v20140518.jar
/WEB-INF/lib/aether-impl-1.0.0.v20140518.jar
/WEB-INF/lib/aether-spi-1.0.0.v20140518.jar
/WEB-INF/lib/aether-util-1.0.0.v20140518.jar
/WEB-INF/lib/aether-connector-basic-1.0.0.v20140518.jar
/WEB-INF/lib/aether-transport-wagon-1.0.0.v20140518.jar
/WEB-INF/lib/maven-aether-provider-3.2.5.jar
/WEB-INF/lib/maven-model-3.2.5.jar
/WEB-INF/lib/maven-model-builder-3.2.5.jar
/WEB-INF/lib/plexus-component-annotations-1.5.5.jar
/WEB-INF/lib/maven-repository-metadata-3.2.5.jar
/WEB-INF/lib/maven-settings-3.2.5.jar
/WEB-INF/lib/maven-settings-builder-3.2.5.jar
/WEB-INF/lib/guava-18.0.jar
/WEB-INF/lib/plexus-interpolation-1.21.jar
/WEB-INF/lib/plexus-utils-3.0.20.jar
/WEB-INF/lib/plexus-sec-dispatcher-1.3.jar
/WEB-INF/lib/plexus-cipher-1.4.jar
/WEB-INF/lib/wagon-provider-api-2.6.jar
/WEB-INF/lib/wagon-file-2.6.jar
/WEB-INF/lib/commons-lang-2.6.jar
/WEB-INF/lib/wagon-http-lightweight-2.6.jar
/WEB-INF/lib/wagon-http-shared-2.6.jar
/WEB-INF/lib/jsoup-1.7.2.jar
/WEB-INF/lib/shrinkwrap-resolver-api-maven-2.2.6.jar
/WEB-INF/lib/shrinkwrap-resolver-api-2.2.6.jar
/WEB-INF/lib/jackson-jaxrs-json-provider-2.8.2.jar
/WEB-INF/lib/jackson-jaxrs-base-2.8.2.jar
/WEB-INF/lib/jackson-core-2.8.2.jar
/WEB-INF/lib/jackson-databind-2.8.2.jar
/WEB-INF/lib/jackson-annotations-2.8.0.jar
/WEB-INF/lib/jackson-module-jaxb-annotations-2.8.2.jar
/WEB-INF/lib/resteasy-client-3.6.2.Final.jar
/WEB-INF/lib/resteasy-jackson-provider-3.1.4.Final.jar
/WEB-INF/lib/jackson-core-asl-1.9.13.jar
/WEB-INF/lib/jackson-mapper-asl-1.9.13.jar
/WEB-INF/lib/jackson-jaxrs-1.9.13.jar
/WEB-INF/lib/jackson-xc-1.9.13.jar
/WEB-INF/lib/weld-servlet-core-2.4.8.Final.jar
/WEB-INF/lib/weld-environment-common-2.4.8.Final.jar
/WEB-INF/lib/weld-core-impl-2.4.8.Final.jar
/WEB-INF/lib/jboss-interceptors-api_1.2_spec-1.0.0.Final.jar
/WEB-INF/lib/weld-spi-2.4.SP2.jar
/WEB-INF/lib/javax.inject-1.jar
/WEB-INF/lib/weld-api-2.4.SP2.jar
/WEB-INF/lib/weld-probe-core-2.4.8.Final.jar
/WEB-INF/lib/weld-core-jsf-2.4.8.Final.jar
/WEB-INF/lib/jboss-el-api_3.0_spec-1.0.0.Alpha1.jar
/WEB-INF/lib/jboss-classfilewriter-1.2.1.Final.jar
/WEB-INF/lib/smallrye-config-1.3.3.jar
/WEB-INF/lib/cdi-api-1.2.jar
/WEB-INF/lib/javax.el-api-3.0.0.jar
/WEB-INF/lib/javax.interceptor-api-1.2.jar
/WEB-INF/lib/javax.annotation-api-1.2.jar
```

--- a/pom.xml
+++ b/pom.xml
@@ -29,7 +29,7 @@
     <version.opentracing>0.31.0</version.opentracing>
     <version.opentracing.jaxrs>0.3.1</version.opentracing.jaxrs>
     <version.opentracing.interceptors>0.0.4</version.opentracing.interceptors>
-    <version.javax.ws.rs-api>2.0</version.javax.ws.rs-api>
+    <version.javax.ws.rs-api>2.1</version.javax.ws.rs-api>
     <version.resteasy>3.6.2.Final</version.resteasy>
     <version.weld>2.4.7.Final</version.weld>
     <version.wildfly>13.0.0.Final</version.wildfly>
@@ -85,6 +85,12 @@
         <groupId>io.opentracing.contrib</groupId>
         <artifactId>opentracing-interceptors</artifactId>
         <version>${version.opentracing.interceptors}</version>
+        <exclusions>
+          <exclusion>
+            <groupId>javax</groupId>
+            <artifactId>javaee-api</artifactId>
+          </exclusion>
+        </exclusions>
       </dependency>
 
       <dependency>

